Planning time failure: Streaming aggregation with input not grouped on the grouping keys

To reproduce, add the following to `TestHiveDistributedQueries`:

```
    @Test
    public void test()
    {
        assertQuery("SELECT\n" +
                "      linenumber,\n" +
                "      'xxx'\n" +
                "  FROM\n" +
                "  (\n" +
                "      (SELECT orderkey, linenumber FROM lineitem)\n" +
                "      UNION\n" +
                "      (SELECT orderkey, linenumber FROM lineitem)\n" +
                "  ) WHERE orderkey = 1 \n" +
                "  GROUP BY\n" +
                "      1");
    }
```

```
Caused by: java.lang.IllegalArgumentException: Streaming aggregation with input not grouped on the grouping keys
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:135)
	at com.facebook.presto.sql.planner.sanity.ValidateStreamingAggregations$Visitor.visitAggregation(ValidateStreamingAggregations.java:89)
	at com.facebook.presto.sql.planner.sanity.ValidateStreamingAggregations$Visitor.visitAggregation(ValidateStreamingAggregations.java:52)
	at com.facebook.presto.sql.planner.plan.AggregationNode.accept(AggregationNode.java:200)
	at com.facebook.presto.sql.planner.plan.InternalPlanNode.accept(InternalPlanNode.java:31)
	at com.facebook.presto.sql.planner.sanity.ValidateStreamingAggregations$Visitor.lambda$visitPlan$0(ValidateStreamingAggregations.java:73)
```

CC: @highker @wenleix @rongrong @hellium01

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java
@@ -4086,6 +4086,22 @@ public class TestHiveIntegrationSmokeTest
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testGroupByWithUnion()
+    {
+        assertQuery("SELECT\n" +
+                "      linenumber,\n" +
+                "      'xxx'\n" +
+                "  FROM\n" +
+                "  (\n" +
+                "      (SELECT orderkey, linenumber FROM lineitem)\n" +
+                "      UNION\n" +
+                "      (SELECT orderkey, linenumber FROM lineitem)\n" +
+                "  ) WHERE orderkey = 1 \n" +
+                "  GROUP BY\n" +
+                "      linenumber");
+    }
+
     private HiveInsertTableHandle getHiveInsertTableHandle(Session session, String tableName)
     {
         Metadata metadata = ((DistributedQueryRunner) getQueryRunner()).getCoordinator().getMetadata();

--- a/presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java
+++ b/presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java
@@ -99,7 +99,6 @@ import static com.facebook.presto.sql.planner.optimizations.ActualProperties.Glo
 import static com.facebook.presto.sql.planner.optimizations.ActualProperties.Global.partitionedOn;
 import static com.facebook.presto.sql.planner.optimizations.ActualProperties.Global.singleStreamPartition;
 import static com.facebook.presto.sql.planner.optimizations.ActualProperties.Global.streamPartitionedOn;
-import static com.facebook.presto.sql.planner.optimizations.AddExchanges.computeIdentityTranslations;
 import static com.facebook.presto.sql.planner.optimizations.AddExchanges.toVariableReference;
 import static com.facebook.presto.sql.relational.OriginalExpressionUtils.castToExpression;
 import static com.facebook.presto.sql.relational.OriginalExpressionUtils.isExpression;
@@ -621,7 +620,7 @@ public class PropertyDerivations
         {
             ActualProperties properties = Iterables.getOnlyElement(inputProperties);
 
-            Map<VariableReferenceExpression, VariableReferenceExpression> identities = computeIdentityTranslations(node.getAssignments(), types);
+            Map<VariableReferenceExpression, VariableReferenceExpression> identities = computeIdentityTranslations(node.getAssignments().getMap(), types);
 
             ActualProperties translatedProperties = properties.translate(column -> Optional.ofNullable(identities.get(column)));
 
@@ -777,6 +776,17 @@ public class PropertyDerivations
 
             return Optional.of(ImmutableList.copyOf(builder.build()));
         }
+    }
+
+    private static Map<VariableReferenceExpression, VariableReferenceExpression> computeIdentityTranslations(Map<VariableReferenceExpression, Expression> assignments, TypeProvider types)
+    {
+        Map<VariableReferenceExpression, VariableReferenceExpression> inputToOutput = new HashMap<>();
+        for (Map.Entry<VariableReferenceExpression, Expression> assignment : assignments.entrySet()) {
+            if (assignment.getValue() instanceof SymbolReference) {
+                inputToOutput.put(toVariableReference(Symbol.from(assignment.getValue()), types), assignment.getKey());
+            }
+        }
+        return inputToOutput;
     }
 
     static boolean spillPossible(Session session, JoinNode.Type joinType)

